app/factories/bulkrax/object_factory.rb

```diff
@@ -166,7 +166,7 @@ def find_by_id
 
     def delete(_user)
       obj = find
-      return false unless obj
+      raise ObjectFactoryInterface::ObjectNotFoundError, "Object not found to delete" unless obj
```

Comment:
It is misleading for a deletion to not happen but be reported as deleted. This ensures that the one running the importer knows about the failure.